batch/batch/worker/worker.py

```diff
@@ -1610,7 +1578,7 @@ def __init__(
         self.jar_url = self.user_command_string[2]
 
         self.deleted = False
-        self.timings = Timings(lambda: self.deleted)
+        self.timings = Timings()
```

Comment:
I think this effectively removes the ability to stop `JVMJob`s mid-execution. With the existing functionality, every `with self.step` in the `run` method is an opportunity to raise a `JobDeletedException` and skip to the cleanup step. As it is now, I don't see a way in which `run` can be interrupted when the job is cancelled. I think that in order to eliminate this, we need to make the `run` method look more like the `Container.run` method, which uses `run_until_done_or_deleted` to allow for deletion mid-run.

Reply:
I think my new changes actually enable this behavior. I don't think it was there before.

Reply:
I'm confused, maybe we're not talking about the same thing? I'm saying that stopping `run` between steps worked here when `Timings` still threw a `JobDeletedError`. The more fine-grained approach that you're implementing here is definitely better though, but that didn't exist when I wrote the top comment.

Reply:
Oh I see. I was looking at `__aexit__` and not `__aenter__` where the exception is raised in both.